src/quacc/atoms/skzcam.py

```diff
@@ -13,12 +13,598 @@
+    -------
+    BlockInfo
+        The ORCA input block (to be put in 'orcablocks' parameter) as a string for the adsorbate-slab complex, the adsorbate, and the slab in a dictionary with the keys 'adsorbate_slab', 'adsorbate', and 'slab' respectively.
+    """
+
+    # First generate the preamble block
+    preamble_block = generate_orca_input_preamble(
+        embedded_cluster=embedded_adsorbed_cluster,
+        quantum_cluster_indices=quantum_cluster_indices,
+        element_info=element_info,
+        pal_nprocs_block=pal_nprocs_block,
+        method_block=method_block,
+        scf_block=scf_block,
+    )
+
+    # Generate the coords block
+    coords_block = generate_coords_block(
+        embedded_adsorbed_cluster=embedded_adsorbed_cluster,
+        quantum_cluster_indices=quantum_cluster_indices,
+        ecp_region_indices=ecp_region_indices,
+        ecp_info=ecp_info,
+        include_cp=include_cp,
+        multiplicities=multiplicities,
+    )
+
+    # Combine the blocks
+    return {
+        "adsorbate_slab": preamble_block + coords_block["adsorbate_slab"],
+        "adsorbate": preamble_block + coords_block["adsorbate"],
+        "slab": preamble_block + coords_block["slab"],
+    }
+
+
+def create_atom_coord_string(
+    atom: Atom,
+    ghost_atom: bool = False,
+    atom_ecp_info: str | None = None,
+    pc_charge: float | None = None,
+) -> str:
+    """
+    Creates a string containing the Atom symbol and coordinates in the ORCA input file format, with additional information for atoms in the ECP region as well as ghost atoms.
+
+    Parameters
+    ----------
+    atom
+        The ASE Atom (not Atoms) object containing the atomic coordinates.
+    atom_ecp_info
+        If not None, then assume this is an atom in the ECP region and adds the ECP info.
+    ghost_atom
+        If True, then the atom is a ghost atom.
+    pc_charge
+        The point charge value for the ECP region atom.
+
+    Returns
+    -------
+    str
+        The atom symbol and coordinates in the ORCA input file format.
+    """
+
+    # If ecp_info is not None and ghost_atom is True, raise an error
+    if atom_ecp_info and ghost_atom:
+        raise ValueError("ECP info cannot be provided for ghost atoms.")
+
+    # Check that pc_charge is a float if atom_ecp_info is not None
+    if atom_ecp_info and pc_charge is None:
+        raise ValueError("Point charge value must be given for atoms with ECP info.")
+
+    if ghost_atom:
+        atom_coord_str = f"{(atom.symbol + ':').ljust(3)} {' '*16} {atom.position[0]:-16.11f} {atom.position[1]:-16.11f} {atom.position[2]:-16.11f}\n"
+    elif atom_ecp_info is not None:
+        atom_coord_str = f"{(atom.symbol + '>').ljust(3)} {pc_charge:-16.11f} {atom.position[0]:-16.11f} {atom.position[1]:-16.11f} {atom.position[2]:-16.11f}\n{atom_ecp_info}"
+    else:
+        atom_coord_str = f"{atom.symbol.ljust(3)} {' '*16} {atom.position[0]:-16.11f} {atom.position[1]:-16.11f} {atom.position[2]:-16.11f}\n"
```

Comment:
Can you remind me: is there any reason to do the formatting instead of just doing `atom.position[0]` as-is, for instance?

Reply:
I have chosen to use this formatting as it makes it easier to read the positions since the x, y and z will appear as a column for all of the atoms. There is no control over the length of the number with atom.position[0], the number of decimal places could be anywhere from 1dp to 10dp, making it quite hard to look at the atom positions.

Reply:
Fair enough!